### PR TITLE
#168734603 add the amenity domain + entity + service models

### PR DESCRIPTION
**Pull request by vitrv**
# add the amenity domain + entity + service models

## reviewer

- @fredbelotte

## contributing

this project is maintained under the terms of the [Contribution Guidelines][contribution-guidelines-url].

[contribution-guidelines-url]: https://github.com/revaturexyz/amenityxyz/blob/master/.github/CONTRIBUTING.md 'CONTRIBUTION GUIDELINES'

### CODE REVIEW

**Comment by fredbelotte** [comment]
the pull request documentation is not filed out...

**Comment by fredbelotte** [comment]
why the assignee not assigned?

**Comment by Smith-Justin** [comment]
Fixed documentation and added assignees.

File: aspnet/Xyz.Amenity.Api/Models/ApiAmenityModel.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using Xyz.Amenity.Lib.Models;
+
+namespace Xyz.Amenity.Api.Models
+{
+    public class ApiAmenityModel : AmenityModel
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Api/Models/ApiAmenityModel.cs]: what is this???

File: aspnet/Xyz.Amenity.Dbx/Models/EntityAmenityModel.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System;
+using Xyz.Amenity.Lib.Models;
+
+namespace Xyz.Amenity.Dbx.Models
+{
+    public class EntityAmenityModel : AmenityModel
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Dbx/Models/EntityAmenityModel.cs]: how is this following naming convention?

File: aspnet/Xyz.Amenity.Lib/Abstracts/ABaseAmenity.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+
+namespace Xyz.Amenity.Lib.Abstracts
+{
+    public abstract class ABaseAmenity
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Lib/Abstracts/ABaseAmenity.cs]: what are we trying to achieve here?

File: aspnet/Xyz.Amenity.Lib/Factory/AmenityCreator.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+using Xyz.Amenity.Lib.Abstracts;
+using Xyz.Amenity.Lib.Models;
+
+namespace Xyz.Amenity.Lib.Factory
+{
+    public class AmenityCreator : AAmenityFactory
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Lib/Factory/AmenityCreator.cs]: naming convention???

File: aspnet/Xyz.Amenity.Lib/Models/AmenityModel.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using Xyz.Amenity.Lib.Abstracts;
+
+namespace Xyz.Amenity.Lib.Models
+{
+    //Test comment
+    public class AmenityModel : ABaseAmenity
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Lib/Models/AmenityModel.cs]: see earlier? why abstract and than rewrite most of this again in the inherited class???

**Pull request closed without merging**